<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MazeScene</class>
 <widget class="QMainWindow" name="MazeScene">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>520</width>
    <height>680</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>520</width>
    <height>680</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>520</width>
    <height>680</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Maze</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="s21::MazeField" name="field" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>500</width>
      <height>500</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: white</string>
    </property>
   </widget>
   <widget class="QPushButton" name="loadPushButton">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>550</y>
      <width>320</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Load the maze</string>
    </property>
   </widget>
   <widget class="QPushButton" name="generatePushButton">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>645</y>
      <width>200</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Generate the maze</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>645</y>
      <width>200</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Solve the maze</string>
    </property>
   </widget>
   <widget class="QLabel" name="startPointLabel">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>580</y>
      <width>90</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="text">
     <string>Start point</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="yStartPointLabel">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>620</y>
      <width>25</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Y =</string>
    </property>
   </widget>
   <widget class="QLabel" name="xStartPointLabel">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>600</y>
      <width>25</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>X =</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="yStartPointSpinBox">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>620</y>
      <width>25</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::NoButtons</enum>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="xStartPointSpinBox">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>600</y>
      <width>25</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::NoButtons</enum>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
   </widget>
   <widget class="QLabel" name="xEndPointLabel">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>600</y>
      <width>25</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>X =</string>
    </property>
   </widget>
   <widget class="QLabel" name="endPointLabel">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>580</y>
      <width>90</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="text">
     <string>End point</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QSpinBox" name="xEndPointSpinBox">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>600</y>
      <width>25</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::NoButtons</enum>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="yEndPointSpinBox">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>620</y>
      <width>25</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::NoButtons</enum>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
   </widget>
   <widget class="QLabel" name="yEndPointLabel">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>620</y>
      <width>25</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Y =</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="rowsSpinBox">
    <property name="geometry">
     <rect>
      <x>396</x>
      <y>600</y>
      <width>25</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::NoButtons</enum>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="columnsSpinBox">
    <property name="geometry">
     <rect>
      <x>396</x>
      <y>620</y>
      <width>25</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="buttonSymbols">
     <enum>QAbstractSpinBox::NoButtons</enum>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
   </widget>
   <widget class="QLabel" name="columnsLabel">
    <property name="geometry">
     <rect>
      <x>326</x>
      <y>620</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Columns</string>
    </property>
   </widget>
   <widget class="QLabel" name="rowsLabel">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>600</y>
      <width>41</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Rows</string>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21::MazeField</class>
   <extends>QWidget</extends>
   <header>view/maze_view/maze_field.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
